server/api/crud/pipelines.py

```diff
@@ -363,6 +371,18 @@ def _resolve_project_from_command(
     def resolve_project_from_pipeline(self, pipeline: PipelineRun):
         return self.resolve_project_from_workflow_manifest(pipeline.workflow_manifest())
 
+    def get_error_from_pipeline(self, kfp_client, run: PipelineRun):
+        detail = kfp_client._run_api.get_run(run.id)
```

Comment:
The whole idea of pipeline_adapter is that we leave the logic of retrieving its internal information to it and not performing any kfp related logic on mlrun side
so, accessing onto internals is an anti pattern
create the logic on both v1.8/v2 adapters where you provide a client and get the run according to its relevant api